### ISSUE TITLE
Switch from cjs to esm

### ISSUE THREAD

**Issue by viceice** [renovatebot/renovate#9890]
**What would you like Renovate to be able to do?**
As more and more dependencies are only available as esm we should also start transitioning to esm.

Full esm is supported since node [v12.22](https://nodejs.org/dist./v12.22.0/docs/api/esm.html#esm_modules_ecmascript_modules), v14 has [no stable esm](https://nodejs.org/dist./v14.16.0/docs/api/esm.html#esm_modules_ecmascript_modules)

**Did you already have any implementation ideas?**

1. First start publishing renovate as cjs and esm, so downstream projects can start testing esm too.
2. Remove cjs from renovate

*Blocker**
- https://github.com/facebook/jest/pull/10976

**Comment by viceice** [comment]
![image](https://user-images.githubusercontent.com/1798109/117301370-e1a71a00-ae7a-11eb-8b0a-e20fe9f817f8.png)
https://nodejs.org/api/esm.html#esm_modules_ecmascript_modules

**Comment by viceice** [comment]
v14 is esm stable too, but docs update not yet released

ref: nodejs/node#38409 nodejs/node#38480

**Comment by thernstig** [comment]
> Full esm is supported since node [v12.22](https://nodejs.org/dist./v12.22.0/docs/api/esm.html#esm_modules_ecmascript_modules), v14 has [no stable esm](https://nodejs.org/dist./v14.16.0/docs/api/esm.html#esm_modules_ecmascript_modules)

I just happened to stumble upon this, and wanted to be a nice citizen 😄 

v14 does have stable ESM support, that is just a bug in the docs. See https://github.com/nodejs/node/issues/38409

**Comment by voxpelli** [comment]
Was this solved by #9887 or additional work needed? Found some comments in #11133 around ESM that confused me. Is ESM modules considered not usable by Renovate yet or?

**Comment by viceice** [comment]
See description of issue, this is blocked by jest issue.

**Comment by HonkingGoose** [comment]
@viceice would having this in Jest be enough to get us going?

https://github.com/facebook/jest/pull/11818

**Comment by viceice** [comment]
> @viceice would having this in Jest be enough to get us going?
> 
> [facebook/jest#11818](https://github.com/facebook/jest/pull/11818)

I don't think so, it's very basic and experimental. Lets wait for more stabillity.

**Comment by rarkins** [comment]
Blocked by #13181
